src/veh_interact.cpp

```diff
@@ -1787,12 +1826,21 @@ void veh_interact::display_stats()
     fold_and_print( w_stats, y[5], x[5], w[5], c_ltgray, wheel_state_description( *veh ).c_str() );
 
     // Write the most damaged part
+    int mostDamagedPart = get_most_damaged_part();
+    int most_repariable = get_most_repariable_part();
     if (mostDamagedPart != -1) {
-        std::string partName;
-        mvwprintz(w_stats, y[6], x[6], c_ltgray, _("Most damaged:"));
-        const auto iw = utf8_width(_("Most damaged:")) + 1;
-        x[6] += iw;
-        w[6] -= iw;
+        if( mostDamagedPart == most_repariable){
+            mvwprintz(w_stats, y[6], x[6], c_ltgray, _("Most damaged:"));
+            auto iw = utf8_width(_("Most damaged:")) + 1;
```

Comment:
The string shouldn't be duplicated. A temp. variable could help here.
Same below.